### PR TITLE
Remove `recursion_limit` config option but limit length of messages

### PR DESCRIPTION
**Pull request by charl3sj**
Ref https://github.com/rnadigital/agentcloud/issues/520

### CODE REVIEW

File: webapp/src/migrations/1.9.0.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import debug from 'debug';
+const log = debug('webapp:migration:1.9.0');
+
+export default async function (db) {
+	log('renaming chatAppConfig.recursionLimit to maxMessages');
+	await db.collection('apps').updateMany(
+		{},
+		{
+			$rename: {
+				'chatAppConfig.recursionLimit': 'chatAppConfig.maxMessages'
```
Comment (tomlynchRNA) [inline webapp/src/migrations/1.9.0.ts:10]: @charl3sj can you update package.json version to 1.9.0 to correspond with this? Good job on already making the migration :smile:

**Comment by tomlynchRNA** [comment]
@charl3sj  Here are two runs with max messages set to 5:

image
(more than 5 messages)


image
(less than 5 messages)

I added some logs and in the first example, if you then ask a question that uses a tool and streams, it does stop the chat, and my `len(past_messages)` was 26. So in the first example, I think it's missing a check for max_messages somewhere.

In the second run, I think it's counting some messages that are not sent to the frontend?

**Review by tomlynchRNA — CHANGES_REQUESTED** [review]
See comment :point_up:

**Comment by charl3sj** [comment]
> In the second run, I think it's counting some messages that are not sent to the frontend?

Yep, there are more messages in state than are shown in the frontend. I moved the limit checking logic to a function [here](https://github.com/rnadigital/agentcloud/pull/521/commits/5eb0dd9e6633b3f6b8179cefec1c19df5f335424#diff-3a034cc0b56e9a5b1dcb7fee42141f947bb1e9cbefc084c3b4dcd9b98c212bddR86) that we can tweak. Also, moving the check to [here](https://github.com/rnadigital/agentcloud/pull/521/commits/5eb0dd9e6633b3f6b8179cefec1c19df5f335424#diff-3a034cc0b56e9a5b1dcb7fee42141f947bb1e9cbefc084c3b4dcd9b98c212bddR152) can yield more precise behavior. Leaving it where it is [now](https://github.com/rnadigital/agentcloud/pull/521/commits/5eb0dd9e6633b3f6b8179cefec1c19df5f335424#diff-3a034cc0b56e9a5b1dcb7fee42141f947bb1e9cbefc084c3b4dcd9b98c212bddR101) allows for some latitude in the limit check in that it waits for one loop iteration to end so that it doesn't stop in the middle of a response.

**Review by tomlynchRNA — APPROVED** [review]

**Pull request merged**